samples/basic-nginx/Jenkinsfile

```diff
@@ -0,0 +1,147 @@
+library identifier: "pipeline-library@1.6",
+retriever: modernSCM(
+  [
+    $class: "GitSCMSource",
+    remote: "https://github.com/redhat-cop/pipeline-library.git"
+  ]
+)
+
+
+openshift.withCluster() {
+  env.APP_NAME = "${JOB_NAME}".replaceAll(/-build.*/, '')
+  env.ARTIFACT_DIRECTORY = "samples/basic-nginx/build"
+  env.BUILD = openshift.project()
+  env.DEV = "${APP_NAME}-dev"
+  env.STAGE = "${APP_NAME}-stage"
+  env.PROD = "${APP_NAME}-prod"
+  env.PELORUS = "pelorus"
+  env.WEBHOOK_APP_NAME = "build-webhook"
+  echo "Starting Pipeline for ${APP_NAME}..."
+}
+
+pipeline {
+  agent {
+    label 'maven'
+  }
+
+  stages {
+    
+    stage('Git Checkout') {
+      steps {
+        git url: "${SOURCE_REPOSITORY_URL}", branch: "${SOURCE_REPOSITORY_REF}"
+        sh "echo ${GIT_COMMIT}"
+      }
+    }
+
+    stage('Build') {
+      steps {
+        script {
+          sh """
+              cd samples/basic-nginx
+              mkdir build
+              cp index.html build/index.html
+              cp nginx.conf build/nginx.conf
+            """
+        }
+      }
+    }
+
+    stage('Image Build') {
+      steps {
+        binaryBuild(projectName: "${BUILD}", buildConfigName: "${APP_NAME}", buildFromPath: "${ARTIFACT_DIRECTORY}")
+      }
+    }
+
+    stage('Send Data to Pelorus') {
+      steps {
+        script {
+          openshift.withCluster() {
```

Comment:
@shaheinm this is the section in the Jenkinsfile that is reaching out to OCP to gather the image sha, webhook route, and webhook secret. Likely in a real deployment, the webhook route and secret would be defined elsewhere as env vars / jenkins credentials. Gathering the image sha will depend on where teams are pushing the image being built within the pipeline.